System/Console/Haskeline.hs

```diff
@@ -244,8 +245,7 @@ getPassword x = promptedInput
                                         $ runCommandLoop tops prefix loop
                                         $ Password [] x)
                     (\fops -> let h_in = inputHandle fops
-                              in bracketSet (hGetEcho h_in) (hSetEcho h_in) False
-                                  $ runMaybeT $ getLocaleLine fops)
+                              in withoutInputEcho h_in $ runMaybeT $ getLocaleLine fops)
```

Comment:
Were you able to make sense of my comment from: https://github.com/judah/haskeline/issues/50#issuecomment-271436812 about integrating this with FileOps and Backend?  I think it would help reduce complexity around the code splits between the Posix and Windows backends.